packages/wrangler/src/module-collection.ts

```diff
@@ -123,9 +123,9 @@ export default function createModuleCollector(props: {
 						{
 							filter: new RegExp(
 								"^(" +
-									[...props.wrangler1xlegacyModuleReferences.fileNames].join(
-										"|"
-									) +
+									[...props.wrangler1xlegacyModuleReferences.fileNames]
+										.join("|")
+										.replace(/[*+?^${}()|[\]\\]/g, "\\$&") +
```

Comment:
I feel like this replacement should be done before joining the fileNames, since the `|` in the join will get escaped?

Reply:
Good catch, have made that change